src/components/inputs/checkbox-input/checkbox.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import vars from '../../../../materials/custom-properties';
+
+// accessible input :)
+const Input = styled.input`
+  pointer-events: none;
+  height: 100%;
+  left: 0;
+  opacity: 0.0001;
+  position: absolute;
+  top: 0;
+  width: 100%;
+
+  ${props =>
+    !props.hasError &&
+    `
+  &:focus + div > svg [id$='borderAndContent'] > [id$='border'] {
+    stroke: ${vars.borderColorInputFocus};
+  }`}
+`;
+
+class Checkbox extends React.Component {
+  static displayName = 'Checkbox';
+
+  static propTypes = {
+    isIndeterminate: PropTypes.bool,
+  };
+
+  componentDidMount() {
+    if (this.props.isIndeterminate) {
+      this.ref.current.indeterminate = true;
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.isIndeterminate !== this.props.isIndeterminate) {
+      this.ref.current.indeterminate = this.props.isIndeterminate;
```

Comment:
for some reason in dev tools you don't actually see the property in the dom, but if you change the opacity of the input itself, you can see that it is treated properly 👍